lib/backend/k8s/resources/node_conversion.go

```diff
@@ -24,23 +24,29 @@ import (
 	kapiv1 "k8s.io/client-go/pkg/api/v1"
 )
 
+// Convert a Kubernetes format node, with Calico annotations, to a Calico Node
 func K8sNodeToCalico(node *kapiv1.Node) (*model.KVPair, error) {
-	// Get the internal IP of the node, should be 1-1 with calico node IP
-	nodeIP := ""
-	for _, address := range node.Status.Addresses {
-		if address.Type == kapiv1.NodeInternalIP {
-			nodeIP = address.Address
-			log.Debugf("Found NodeInternalIP %s", nodeIP)
-			break
-		}
+	annotations := node.ObjectMeta.Annotations
+
+	ip := net.ParseIP(annotations["projectcalico.org/IPv4"])
+
+	_, bgpV4Cidr, err := net.ParseCIDR(annotations["projectcalico.org/BGPIPv4Net"])
+	if err != nil {
+		log.Warnf("Could not parse BGPIPv4CIDR from k8s annotation: %s",
+			annotations["projectcalico.org/BGPIPv4Net"])
+	}
+
+	bgpAsn, err := numorstring.ASNumberFromString(annotations["projectcalico.org/BGPASNumber"])
+	if err != nil {
+		log.Warnf("Could not parse ASNumber from k8s annotation: %s",
+			annotations["projectcalico.org/BGPASNumber"])
```

Comment:
If we pull the values out of the annotations at the top, then we don't have to keep accessing them from the map like this all over.  I think I'd like that better.